sqlglot/expressions.py

```diff
@@ -484,43 +484,78 @@ class Interval(Expression):
 # Functions
 class Func(Expression):
     token_type = TokenType.FUNC
+    ordered_arg_types = [('this', True)]
+    is_var_len_args = False
+
+    def __init__(self, **args):
+        self.arg_types = dict(self.ordered_arg_types)
```

Comment:
it was designed specifically to have arg_types so that we don't have to worry about different dialects having different orders in functions and that with kwargs it's very explicit what each arg is, it looks like this change is making all functions use ordered_arg_types? 

i still think regular args are better and that ordered_arg_types should be optional

Reply:
Makes sense. Will update this.

Reply:
Simplified the implementation as per our conversation earlier.